### ISSUE TITLE
New Feature Request: Vertical Table Titles

### ISSUE THREAD

**Issue by ordric** [zim-desktop-wiki/zim-desktop-wiki#1352]
Would it be possible to add the ability to create vertical table headers. This is possible in DokuWiki, on which Zim is based, by using the caret ('^') to signify the header and the vertical bar ('|') for non header cells

For Example:

```
^ Header 1 ^ Header 2 ^ Header 3 |
|                  |                   |                  |
|                  |                   |                  |
|                  |                   |                  |
Would produce a table with horizontal headers.

^ Header 1 |                  |                  |
^ Header 2 |                  |                  |
^ Header 3 |                  |                  |
Would produce a table with vertical headers.

|                  | ^ Header 1 ^ Header 2 |
^ Header 3 |                    |                  |
^ Header 4 |                    |                  |
^ Header 5 |                    |                  |
Would produce a table with both horizontal and vertical headers.
```
Would this be possible please?

It would certainly be very helpful.

**Comment by pfalcon** [comment]
> This is possible in DokuWiki, on which Zim is based

What makes you think that Zim is "based" on DokuWiki, and what do you mean by "based" at all?

**Comment by ordric** [comment]
Sorry, I should have said 'inspired by', but the essence is the same:

"Zim - A Desktop Wiki

Syntax

Examples of the wiki syntax as used by zim.

Most of this syntax is inspired by the DokuWiki syntax, but with some significant modifications. All markup uses at least two characters to reduce the chance of conflicts between the wiki markup normal typographic use of these characters."
[WIKI Syntax](https://zim-wiki.org/manual/Help/Wiki_Syntax.html)

**Comment by pfalcon** [comment]
Thanks, that explains it.

Next step probably would be to explain what you mean by "*table headers*". If you mean look of the GTK widget, which Zim happens to currently use for table representation *during editing*, that may be a tough luck to expect it to show shaded-gradient "headers" on the left. But if you mean a common rendered look of tables, where headers are just represented with bold font, then do just that - put you headers anywhere you want, and add them bold formatting, problem solved.

**Comment by ordric** [comment]
No, since I was talking about the Zim Syntax, and gave the examples from the DokuWiki syntax, I assumed it was implicit that I was asking if the syntax, not the plugin, could be enhanced to incorporate this. 

Just putting bold entries as headers where I want simply doesn't work, and it isn't 'problem solved'. For example, if I put the following coding into a page's text file:

```
| **Entry 1** | Entry 4 | Entry 5 |
| **Entry 2** |         |         | 
| **Entry 3** |         | Entry 6 | 
```
I should get a nice set of emboldened headers in the first column and an OK table, according to you. What I actually get is this:

![image](https://user-images.githubusercontent.com/41547625/103535329-6b70b780-4e88-11eb-9183-363e5b9fb742.png)

As you can see, the top row is always a header row in Zim whether I like it or not, and header rows don't like bold/italic coding.

Now as I asked originally, if it were possible to enhance the syntax, to identify the header cells of a table and shade/embolden or whatever, where they are defined, that would be great.

**Comment by pfalcon** [comment]
> I should get a nice set of emboldened headers in the first column and an OK table, according to you.

No, I didn't say that.

> Now as I asked originally, if it were possible to enhance the syntax, to identify the header cells of a table and shade/embolden or whatever, where they are defined, that would be great.

But you exactly don't ask about *syntax*, you ask about *rendering* that syntax *during editing*, and that's the hard part.

And what I talk about is - make cells on the left bold, press Ctrl+P, and behold your headers at the top and left, rendered in similar way (bold font, again).

**Comment by ordric** [comment]
**_"But you exactly don't ask about syntax, you ask about rendering that syntax during editing, and that's the hard part."_**

I don't want to split hairs but I 'exactly' did ask that and nothing else. I wrote:

**_"Would it be possible to add the ability to create vertical table headers. This is possible in DokuWiki, on which Zim is based, by using the caret ('^') to signify the header and the vertical bar ('|') for non header cells"_**
  
And I gave examples from DokuWiki of how their syntax works, I said nothing about rendering while editing. 

Since the syntax is then rendered, of course this means that the render engine will need to be altered to interpret the syntax enhancement. But the 'feature request' I asked for was an 'ehancement' of the syntax, not an enhancement to Zim's plugin editor.

**_"And what I talk about is - make cells on the left bold, press Ctrl+P, and behold your headers at the top and left, rendered in similar way (bold font, again)."_**

Whis is not what you said originally, and not what I asked for. You said:

**_"But if you mean a common rendered look of tables, where headers are just represented with bold font, then do just that - put you headers anywhere you want, and add them bold formatting, problem solved."_**

Which is not what I was asking for. That is editing in Zim. I asked for an enhancemement to the syntax and the concomitant renderer being updated to be able to interpret that syntax enhancement, so that syntax in the file will be rendered with headers as I asked. That is all.

Perhaps it would work better, if you imagine I have written my own shellscript and when I operate it, it inserts it's output into the Zim page. In this way, I need to be able to instruct my shellscript to use the syntax according to the enhanced syntax and the renderer simply interprets syntax as it is loaded. 

It could be achieved easier than the DokuWiki I examples I gave at the top by, say altering the renderer to interpret an alteration of the first character in the table syntax for example:

```
| Entry 1 | Entry 4 | Entry 5 |
| Entry 2 |         |         | 
| Entry 3 |         | Entry 6 |
```
Would be interpreted by the renderer as syntax for a horizontal header, while:

```
_ Entry 1 | Entry 4 | Entry 5 |
| Entry 2 |         |         | 
| Entry 3 |         | Entry 6 |
```
Would be interpreted as a table with vertical headers. This would ensure that the in-built table editor plugin would continue to work, while allowing me to use the syntax to create vertical headers as I said at the top, by editing the text file directly. Of course, what character to be used would be up to whoever programmed the enhancement.
